src/messages.rs

```diff
@@ -55,6 +55,13 @@ pub struct ConnectResponse {
     pub connect_request_signature: Signature
 }
 
+/// Respond wiht Data or Error
+#[derive(PartialEq, Eq, Clone, Debug, RustcEncodable, RustcDecodable)]
+pub enum DataOrError {
+Data(Data),
+Err(ResponseError),
+}
```

Comment:
Yea I am thinking this is a mistake here as it really needs `Err(RoutingError, SignedMessage)` where the message is the original message and error states clearly why it failed. I do not think we should ever send success messages (waste of b/w). Only default return is Get where if we have data we return it, Error cases for Get are something to further consider, but not for now.